templates/katello.yaml.erb

```diff
@@ -20,14 +20,16 @@
     :oauth_secret: <%= scope['katello_devel::oauth_secret'] %>
     :ca_cert_file: <%= scope['katello_devel::candlepin_ca_cert'] %>
 
+  :candlepin_events:
+    :broker_host: <%= scope['katello_devel::candlepin_events_broker_host'] %>
+    :broker_port: <%= scope['katello_devel::candlepin_events_broker_port'] %>
+    :queue_name: <%= scope['katello_devel::candlepin_events_queue_name'] %>
+    :subscription_name: <%= scope['katello_devel::candlepin_events_subscription_name'] %>
```

Comment:
This is the pattern we've previously always taken. I've been starting to question this in the present. Are these configurations that need to be done and made prior to runtime? And, do all of them need to be *configurable*?

Reply:
> Are these configurations that need to be done and made prior to runtime?

IMO the app should be told all of these values

> And, do all of them need to be _configurable_?

I don't feel strongly about it and was mostly anticipating 'make these configurable' feedback!

Reply:
I'll break it down a bit further. I see it as two types of configuration:

 1) configuration that the application needs to start **and** that should require an application restart to change
 2) configuration that the application needs to execute workflows but should not break the application if its missing or mis-configured

Given your worker with the listener, it's not clear to me where `broker_host` and `broker_port` fit into this model. The latter two options `queue_name` and `subscription_name` feel like values that ought to be hard-coded rather than configurable by a user.

Reply:
For me it comes down to this question: whose responsibility is it to know the queue name? The listener is not much more than code that subscribes and handles events it receives. I don't think it should know anything about its queue.

My final PR shows how in engine.rb the CandlepinEventListener is given a connection to subscribe on: https://github.com/Katello/katello/pull/8563/files#diff-cf74e596d7887011413ee9d06b16b753R125-R132

Another thing to consider is that the queue and subscription name are in the broker.xml and they should match Katello's configuration. It makes sense to me for their values to originate from somewhere outside the code - a single place. I think even the client_id at the above link should probably be moved out since it also exists in broker.xml.

The presence of a value in a config file is not always for the user's sake. In this case *we* want to configure the connection details.

Reply:
Since my last comment I saw my tests were failing on the katello PR, so I added default settings in engine.rb https://github.com/Katello/katello/pull/8563/files#diff-cf74e596d7887011413ee9d06b16b753R52-R57

This means that the values are still derived from the configuration but we don't need to supply them in dev.

But my earlier point is still relevant - these should be configurable because broker.xml and the config settings need to be in agreement.